client/oslc-client/src/main/java/org/eclipse/lyo/client/query/OslcQueryResult.java

```diff
@@ -124,12 +125,30 @@ private synchronized void initializeRdf() {
 			Property responseInfo = rdfModel.createProperty(OslcConstants.OSLC_CORE_NAMESPACE, "ResponseInfo");
 			ResIterator iter = rdfModel.listResourcesWithProperty(rdfType, responseInfo);
 
-			//There should only be one - take the first
-			infoResource = null;
+			//The main ResposeInfo shall have the query URI or a page URI
+            List<Resource> responseInfos = iter.toList();
+
+            infoResource = null;
+            if (responseInfos.size() == 1) {
+                infoResource = responseInfos.get(0);
+            } else if (responseInfos.size() > 1) {
+                List<Resource> infos_sameURI = responseInfos.stream().filter(ri -> ri.getURI().equals(query.getQueryUrl())).toList();
```

Comment:
would it make more sense to replace "equals" with startsWith?

Reply:
Well, there is no strict rule on the ordering of query params, so the substring check does not have a strong meaning. I was thinking to just cut off all the query params and compare the URLs with all params stripped but I wanted to hear from Ana how her testing goes before altering the URIs.

Reply:
True.

Reply:
Hi Andrew, don't know if you saw it already, but please check my feedback on [the forum](https://forum.open-services.net/t/cannot-obtain-nextpage-url-for-ewm-query-multiple-responseinfo-tags-in-the-query-result/825/8?u=anamr).

Thanks,
Ana

Reply:
@anamrosu, thanks! Sad to hear a direct check did not succeed.

I am thinking of keeping the current checks and adding a substring check as a fallback. 

Unfortunately, I am really busy this week and will not be able to work on the PR (I do not work on Lyo during my day job). Also, we are now in the release freeze period after Jad published the Lyo 6 release candidate. We won't merge any PRs except critical bugfixes until next week.

@anamrosu if you are up for it, I would be glad if you fork the repo, update my code and submit a new PR. Thank you for helping us improve Lyo!

Reply:
@anamrosu just updated the logic, please check.

Reply:
Hi Andrew, sorry for not giving you any updates, I was also pretty busy the past weeks. I will be on vacation starting tomorrow for 2 weeks, so I will test your changes after I come back. Thanks!

Reply:
Hi Andrew, I tested the changes a few days ago and it was still not working for us. So, I thought about an alternative solution which, in my opinion, should work better. I also tested it in our environment and it works.
In this solution I'm trying to directly look for the responseInfo containg "nextPage" in it. There should be exactly one responseInfo matching this case. I created a new pull request with my proposal, please have a look: https://github.com/eclipse-lyo/lyo/pull/646

Reply:
Thanks @anamrosu. I updated my PR. Now the logic goes through the following:

- exactly one ResponseInfo
- otherwise, exactly one ResponseInfo with the nextPage (added in the update)
- otherwise, exactly one ResponseInfo with the exact URI match
- otherwise, exactly one ResponseInfo with the prefix URI match

I could not use your PR directly because:

- you did not sign Eclipse CLA yet: https://api.eclipse.org/git/eca/status/gh/eclipse-lyo/lyo/646 - please create an account with Eclipse and sign the CLA.
- there is a risk that multiple response infos will have the next page (paging is part of OSLC Core, not Query - any valid OSLC Core response may have paging)
- the last page will not have the nextPage but we still need to locate the response info because it contains extra info than just the next page (totalCount, for example - I exposed it in the "sister" PR for oslc4net https://github.com/OSLC/oslc4net/pull/203)
- I wanted to add some logging to help with future improvements - your one-shot iteration was a bit hard to use for that.

Reply:
Hi @berezovskyi, thanks for the detailed explanation and for taking my solution into account when merging the pull request. Do you have an estimate date when these changes will be officially released?